tests/basis_test.py

```diff
@@ -67,6 +67,58 @@ def test_cheb_call_random_points():
         f = ChebyshevFirstKind(n)
         assert numpy.isclose(f(x),special.eval_chebyt(n,x))
 
+
+def test_cheb_derivative():
+    """Test if the correct derivative is generated."""
+    f0 = ChebyshevFirstKind(0)
+    f1 = ChebyshevFirstKind(1)
+    f2 = ChebyshevFirstKind(2)
+    x0, x1 = 1, -0.5
+    assert numpy.isclose(f0.derivative(x0), 0)
```

Comment:
You might get nicer output using `pytest.approx` for these comparisons:
```suggestion
    assert f0.derivative(x0) == pytest.approx(0)
```